modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -312,6 +312,28 @@ public List<ObjectRelationship> getObjectRelationships(
 			objectDefinitionId1, objectDefinition2, type);
 	}
 
+	@Override
+	public List<ObjectRelationship> getObjectRelationships(
+		long objectDefinitionId1, String deletionType, boolean reverse) {
+
+		return objectRelationshipPersistence.findByODI1_DT_R(
+			objectDefinitionId1, deletionType, reverse);
+	}
+
+	@Override
+	public boolean hasDeletionType(
```

Comment:
@marcelabc are you using it just for prevent deletion type, right? Maybe we could implement a method `isPreventDeletionType` with a couple of changes. What do you think?

Reply:
@gabrielwas Okay! We just have to keep in mind to refactor to a generic one if it needs another logic involving another deletionType in the future. 

I created the method this way to make the most generic, but I will change it to a specific one :+1:

Reply:
I rename the method, so I'm removing the label "Author Action Required".

Reply:
After internal discussion, we've decided to keep the method as it is and just rename the method. Thanks, @marcelabc !